internal/engine/compiler/compiler_controlflow_test.go

```diff
@@ -752,6 +752,65 @@ func TestCompiler_compileCallIndirect(t *testing.T) {
 	})
 }
 
+// TestCompiler_callIndirect_largeTypeIndex ensures that non-trivial large type index works well during call_indirect.
+func TestCompiler_callIndirect_largeTypeIndex(t *testing.T) {
+	env := newCompilerEnvironment()
+	table := make([]wasm.Reference, 1)
+	env.addTable(&wasm.TableInstance{References: table})
+	// Ensure that the module instance has the type information for targetOperation.TypeIndex,
+	// and the typeID  matches the table[targetOffset]'s type ID.
+	const typeIndex, typeID = 12345, 0
```

Comment:
Is a large type ID anything more than 8 bits? 12345 seemed smaller than I was expecting :P

Reply:
hehe yeah I think I need a comment why `12345` is considered large here :D